apps/api/src/domain/models/shared/common-commands/provide-additional-credits-for-resource/additional-credits-provided-for-resource.event-handler.spec.ts

```diff
@@ -0,0 +1,157 @@
+        this.arangoDb = new ArangoDatabaseForCollection(
+            new ArangoDatabase(connectionProvider.getConnection()),
+            WIDGET_COLLECTION
+        );
+    }
+
+    async attribute(_id: string, _contributionSummary: ContributionSummary): Promise<void> {
+        throw new Error('Method not implemented.');
+    }
+
+    async fetchById(id: string): Promise<Maybe<WidgetViewModel>> {
+        const searchResult = await this.arangoDb.fetchById(id);
+
+        if (isNotFound(searchResult)) {
+            return searchResult;
+        }
+
+        return new WidgetViewModel(mapDatabaseDocumentToAggregateDTO(searchResult));
+    }
+
+    async create(w: WidgetViewModel): Promise<void> {
+        await this.arangoDb.create(mapEntityDTOToDatabaseDocument(w));
+    }
+}
+
+const existingWidgetView = new WidgetViewModel({
+    id: buildDummyUuid(2),
+});
+
+describe(`AdditionalCreditsProvidedForResourceEventHandler`, () => {
+    let testQueryRepository: IWidgetQueryRepository;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let app: INestApplication;
+
+    let additionalCreditsProvidedForResourceEventHandler: AdditionalCreditsProvidedForResourceEventHandler;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            imports: [PersistenceModule.forRootAsync()],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        testQueryRepository = new WidgetQueryRepository(connectionProvider);
+
+        additionalCreditsProvidedForResourceEventHandler =
+            new AdditionalCreditsProvidedForResourceEventHandler({
+                forResource: (resourceType) => {
+                    if (resourceType !== ('widget' as ResourceType)) {
+                        throw new InternalError(
+                            `this test only supports resources of type 'widget'`
+                        );
+                    }
+
+                    return testQueryRepository;
+                },
+            });
+
+        await connectionProvider.createCollectionIfNotExists(WIDGET_COLLECTION);
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    beforeEach(async () => {
+        await databaseProvider.getDatabaseForCollection(WIDGET_COLLECTION).clear();
+
+        await testQueryRepository.create(existingWidgetView);
+    });
+
+    describe(`when the target is a resource`, () => {
```

Comment:
Let's add a test case that it fails gracefully when the `attribute` method is not implemented on the injected repo.

Reply:
done